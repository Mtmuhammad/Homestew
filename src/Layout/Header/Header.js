import React from "react";
import "./Header.scss"
import {Link} from "react-router-dom"
import braisedSteak from "../../Assets/Images/home/braisedSteak.jpg"
import caesarSalad from "../../Assets/Images/home/caesarSalad.jpg"
import blueberryMuffins from "../../Assets/Images/home/blueberryMuffins.jpg"

const Header = () => {
  return (
    <header style={{ maxWidth: "1100px" }} className="container mt-5">
      <div className="container pt-3 pb-4">
        <h1 className="mb-4">
          Featured Recipes<i className="fa-solid fa-utensils"></i>
        </h1>
        {/* featured recipes */}
        <div className="row">
          <div className="latest-post col-6 col-md-4">
            
            <img
              className="img-fluid"
              src={braisedSteak}
              alt="Braised Steak"
            />
            <div className="caption">
              <Link to="recipes/635874"><p className="caption-text">Braised Steak</p></Link>
            </div>
            
          </div>
          <div className="latest-post col-6 col-md-4">
            
            <img
              className="img-fluid"
              src={caesarSalad}
              alt="Simple Caesar Salad"
            />
            <div className="caption">
              <Link to="recipes/660085"><p className="caption-text">Simple Caesar Salad</p></Link>
            </div>
            
          </div>
          <div className="latest-post col-md-4 d-none d-md-block">
            
            <img
              className="img-fluid"
              src={blueberryMuffins}
              alt="Blueberry Muffins"
            />
            <div className="caption">
            <Link to="recipes/648647"><p className="caption-text">Blueberry Muffins</p></Link>
            </div>
            
          </div>
        </div>
        {/* end featured recipes */}
      </div>
      <hr className="mt-4" />
    </header>
  );
};

export default Header;
