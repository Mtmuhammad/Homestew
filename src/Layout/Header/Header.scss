@use "../../Styles/abstracts/mixins" as *;
@use "../../Styles/abstracts/variables" as *;

header {
  margin-bottom: 3rem;
  & h1 {
    color: $primary-font;
    font-size: 3rem;
    font-weight: 700;
    line-height: 1.1;
    text-transform: uppercase;
    letter-spacing: 0.04em;
    margin: 0;

    & .fa-utensils {
      color: $secondary-color;
      font-size: 4rem;
      margin-left: 2rem;
    }
  }
  & .latest-post {
    position: relative;

    & img {
      border-radius: 5px;
      height: 52rem !important;

      @include respond(port) {
        height: 40rem !important;
      }
    }
  }

  & .caption {
    position: absolute;
    bottom: 25px;
    right: 25px;
    left: 25px;
    display: block;
    padding: 15px 10px;
    text-align: center;
    background-color: rgba(255, 255, 255, 0.8);
    transition: all 0.2s ease-in-out;

    &:hover {
      background-color: #fff;
      cursor: pointer;
    }

    & a {
      text-decoration: none;

      & .caption-text {
        color: $primary-font;
        font-size: 2rem;
        font-weight: 700;
        line-height: 1.1;
        text-transform: uppercase;
        letter-spacing: 0.08em;
        margin: 0;
      }
    }
  }
}
