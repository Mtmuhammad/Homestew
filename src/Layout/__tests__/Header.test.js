import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "../Header/Header";
import {BrowserRouter} from "react-router-dom"


describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<Header /> , {wrapper: BrowserRouter});
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<Header />, {wrapper: BrowserRouter});
    expect(asFragment()).toMatchSnapshot();
  });
});

test("should render h1", () => {
  render(<Header />, {wrapper: BrowserRouter});
  const heading = screen.getByRole("heading", { name: /featured recipes/i });
  expect(heading).toBeInTheDocument();
});

test("should render images", () => {
  render(<Header />, {wrapper: BrowserRouter});
  const images = screen.getAllByRole("img");
  expect(images.length).toBe(3);
  expect(images[0]).toHaveClass("img-fluid");
  expect(images[1]).toHaveClass("img-fluid");
  expect(images[2]).toHaveClass("img-fluid");
});
