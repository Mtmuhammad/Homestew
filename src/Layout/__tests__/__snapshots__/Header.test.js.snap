// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <header
    class="container mt-5"
    style="max-width: 1100px;"
  >
    <div
      class="container pt-3 pb-4"
    >
      <h1
        class="mb-4"
      >
        Featured Recipes
        <i
          class="fa-solid fa-utensils"
        />
      </h1>
      <div
        class="row"
      >
        <div
          class="latest-post col-6 col-md-4"
        >
          <img
            alt="Braised Steak"
            class="img-fluid"
            src="braisedSteak.jpg"
          />
          <div
            class="caption"
          >
            <a
              href="/recipes/635874"
            >
              <p
                class="caption-text"
              >
                Braised Steak
              </p>
            </a>
          </div>
        </div>
        <div
          class="latest-post col-6 col-md-4"
        >
          <img
            alt="Simple Caesar Salad"
            class="img-fluid"
            src="caesarSalad.jpg"
          />
          <div
            class="caption"
          >
            <a
              href="/recipes/660085"
            >
              <p
                class="caption-text"
              >
                Simple Caesar Salad
              </p>
            </a>
          </div>
        </div>
        <div
          class="latest-post col-md-4 d-none d-md-block"
        >
          <img
            alt="Blueberry Muffins"
            class="img-fluid"
            src="blueberryMuffins.jpg"
          />
          <div
            class="caption"
          >
            <a
              href="/recipes/648647"
            >
              <p
                class="caption-text"
              >
                Blueberry Muffins
              </p>
            </a>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="mt-4"
    />
  </header>
</DocumentFragment>
`;
